src/net.cpp

```diff
@@ -936,27 +938,34 @@ void ProtectEvictionCandidatesByRatio(std::vector<NodeEvictionCandidate>& evicti
     size_t num_protected{0};
 
     while (num_protected < max_protect_by_network) {
+        // Count the number of disadvantaged networks from which we have peers to protect.
+        auto num_networks = std::count_if(networks.begin(), networks.end(), [](const Net& n) { return n.count; });
+        if (num_networks == 0) {
+            break;
+        }
```

Comment:
06b898c3e1 `p2p: iterate eviction protection only on networks having candidates`

Looking at this commit - there is no need to re-calculate `num_networks` inside the loop because `networks` is not changed in the loop. It can be calculated just once before the loop:

```cpp
auto num_networks = std::count_if(networks.begin(), networks.end(), [](const Net& n) { return n.count; });
if (networks > 0) {
    while (num_protected < max_protect_by_network) {
        ...
```

I did not observe any performance change due to this commit.

But considering the subsequent ab2993336f `p2p: earlier continuation when no remaining eviction candidates` (this one brings pefromance boost!) it should be inside the loop because then we change `networks[].count` in the loop. Maybe squash both commits into one?

Reply:
Reorganized to be step by step, with no anticipatory changes.